### PR TITLE
Adding new CRUD routes to server

### PR DESCRIPTION
**Pull request by RaneWallin**
**Added:**
POST /traits - add a trait to the database
POST /traits/product - add a trait to a product
Database queries to support the above tasks

**Refactored:**
Moved routes to a separate directory
Refactored original routes to use async/await

**TODO:**
Add tests
Update documentation

### CODE REVIEW

File: db/models/index.js [hunk 1]
```diff
@@ -11,43 +11,57 @@ const NUM_GAMES_PER_TRAIT = 4;
...
     `);
 
-  results = await results.all();
-  console.log(results);
+    results = await results.all();
 
-  return results;
+    return results;
+  } catch (err) {
+    console.log('fetchTraitsForProduct: Error getting products from database.');
```
Comment (margotTenenbaum) [inline db/models/index.js:27]: Might want to include the err in your console.log!

File: package.json [hunk 2]
```diff
@@ -9,10 +9,10 @@
   },
   "scripts": {
     "test": "jest --detectOpenHandles --collectCoverage",
-    "build": "webpack --watch -d",
+    "react:dev": "webpack --watch -d",
     "p_build": "webpack -p",
     "p_start": "node server/server.js",
-    "start": "nodemon server/server.js --ignore node_modules",
+    "server:dev": "nodemon server/index.js --ignore node_modules",
```
Comment (margotTenenbaum) [inline package.json:15]: I like the renaming of scripts for additional clarity!

File: routes/index.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+/* -----------  GET ROUTES --------------------------------------------------------------------------- */
+routes.get('/:product_id', async (req, res) => {
+  try {
+    const id = +req.params.product_id;
+    const traits = await db.fetchTraitsForProduct(id);
+    let traitProducts = [];
+
+    if (!traits || traits.length === 0) {
+      return res.status(404).send('No traits found for this product.');
```
Comment (margotTenenbaum) [inline routes/index.js:18]: Very clear error message.

File: routes/index.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+/* -----------  post ROUTES -------------------------------------------------------------------------- */
+// Add a new trait
+routes.post('/', async (req, res) => {
+  const { trait } = req.body;
+
+  try {
+    let response = await db.addTraitToDatabase(trait);
+
+    if (response === 0) {
+      return res.status(417).send('Could not add trait to database.');
```
Comment (margotTenenbaum) [inline routes/index.js:88]: Good use of 417!

**Comment by margotTenenbaum** [comment]
Approved! Really clear work, easy to follow and great use of error codes and messages.

File: db/models/index.js (hunk 1 above)
Comment (RaneWallin) [reply db/models/index.js:27]: LOL. Good point. Thanks.

**Pull request merged**